### ISSUE TITLE
Nothing shown after installing browser to local machine

### ISSUE THREAD

**Issue by LongTianPy** [epgg/eg#79]
Hi,

I just installed the browser to my local machine following the instructions from http://wiki.wubrowse.org/Install_a_local_mirror, and I also put hg19.gz and hg19.gz.tbi into the seq folder. However, I got nothing when I clicked on the `hg19` link on the index page. It says "Crashed loading genome info", and it would return to the index page with an error message saying "Unknown genome name: hg19".
I'm not sure what is wrong with my installation step, but I was indeed confused at the part of enabling CORS: where should I add the three lines? Inside the <Directory> block of the folder where the browser sits?
Please help me with this issue, thank you!

**Comment by lidaof** [comment]
Hi @LongTianPy , CORS is an Apache setting, please check:
https://poanchen.github.io/blog/2016/11/20/how-to-enable-cross-origin-resource-sharing-on-an-apache-server
https://enable-cors.org/server_apache.html

do you load the makeDb.sql to hg19 too?
Thanks.

**Comment by LongTianPy** [comment]
I think I've set CORS correctly because there has never been an error message about it.  
And yes, I followed every step in the tutorial, including loading makeDb.sql to hg19, so that I have the `config` table in the `hg19` database.

**Comment by LongTianPy** [comment]
Actually, as I'm installing this browser in a Microsoft Azure VM, there might be some dependencies other than those are listed in the tutorial that I didn't install. For example, does it require Perl?

**Comment by lidaof** [comment]
hi @LongTianPy , I don't think Perl is needed. can you do a `git pull` on our repo, on the `config/hg19`, replace the `tracks.json` and `publichub.json` on your server with the ones from the repo?

**Comment by LongTianPy** [comment]
I did `git pull` and replaced the `json` files, and it's still not working. When looking at the previous issues, I found that the apache error log of other people is 
```
subtileKnife starting... >>>>>
cgi param parsed
Hooked to hg19
reviveScaffold() done
```

Whereas mine only shows 
```
subtileKnife starting... >>>>>
cgi param parsed
Hooked to hg19
```

Does it mean that the database is successfully connected but the function of `reviveScaffold` is not run correctly?

**Comment by LongTianPy** [comment]
Although it says `Hooked to hg19`, the error returned on the screen says `Unknown genome name: hg19`. I wonder where these names are retrieved and connected.

**Comment by lidaof** [comment]
ok, weird, would you mind load the database again? under hg19 folder, do 
 cat makeDb.sql | mysql -u hguser -p --local-infile=1 hg19

**Comment by LongTianPy** [comment]
I didn't have a problem loading the database but I did it once again just now. Still doesn't work.

In your `sukn.js`, line 308, `onunknowngenome:on_unknown_genome`, how is this parameter generated? I traced to line 466 of the same script, but I don't understand how it returns 'Unknown genome name'.

**Comment by lidaof** [comment]
Hi @LongTianPy , this is very strange,,actually I never encounter such issue before, I am not familiar with Microsoft Azure VM, are you using Ubuntu?

**Comment by LongTianPy** [comment]
Yes, I'm using Ubuntu 17.04 artful in the VM, only it was built from scratch, that's why I said that I might miss a couple of dependencies, e.g. Java?
Also, is there any method I can check if the website is connected to hg19? When I click other tracks that I haven't implement, they showed the same error message ("Unknown genome name...") as I click hg19. So I guess there's something wrong with the database connection. Do you use any library to access MySQL with Javascript?

**Comment by lidaof** [comment]
the C code subtleKnife.c used mysql.h to bind mysql connection. Java is not needed. I am not sure if it's possible for you to generate a temporary user account for me to debug...as for now I cannot think of any particular reason...really sorry.

**Comment by LongTianPy** [comment]
I can create a user account but this VM can only be accessed within the internal network. But never mind I'll keep working on it. Thank you!

**Comment by lidaof** [comment]
We also have docker images in case you want to give it a try: https://hub.docker.com/r/epgg/eg/
Thanks.

**Comment by LongTianPy** [comment]
Thank you! I've worked it out.
And I did try using the docker image, but MySQL failed to start.

**Comment by lidaof** [comment]
@LongTianPy ok, that;s great! Thanks.
